spring-shell2-core/src/main/java/org/springframework/shell2/standard/StandardParameterResolver.java

```diff
@@ -98,7 +98,7 @@ public void setValueProviders(Collection<ValueProvider> valueProviders) {
 
 	@Override
 	public boolean supports(MethodParameter parameter) {
-		return true;
+		return parameter.getMethodAnnotation(ShellMethod.class) != null;
```

Comment:
Unless I'm missing something, this is wrong and not what is intended.
One aim of the new standard approach is to require the least code possible. No annotation should be required.